pygdf/cudautils.py

```diff
@@ -378,33 +378,36 @@ def compute_scale(arr, vmin, vmax):
 
 
 @cuda.jit
-def gpu_label(arr, cat, val, out):
-    i = cuda.grid(1)
-    if i < out.size:
-        res = (val) if (arr[i] == cat) else arr[i]
-        out[i] = res
+def gpu_label(arr, labarr, out):
+    i, j = cuda.grid(1), cuda.grid(1)
+    for i in range(out.size):
+        val = arr[i]
+        if j < (labarr.size) / 2 and val == labarr[j][0]:
```

Comment:
Use `labarr.shape[0]` instead of `labarr.size/2`

Reply:
Using `labarr.shape[0]` make the size double, since there are two dimensions in the array, which would make an unnecessary iteration for the second time, therefore used `(labarr.size) / 2`.

Reply:
`lab.shape[0] == lab.size/2`

Reply:
btw, the python3 `/` returns float.  Use `//` for floor division to ensure integer return type.

Reply:
ok I will make the change.